src/scripts/Tabs.js

```diff
@@ -3,12 +3,29 @@ import ReactDOM from 'react-dom';
 import classnames from 'classnames';
 import { registerStyle } from './util';
 import DropdownButton from './DropdownButton';
+import { MenuItem } from './DropdownMenu';
 
 export default class Tabs extends Component {
-
   constructor(props) {
     super(props);
-    this.state = {};
+    const visibleTabs = [];
+    const hiddenTabs = [];
+
+    props.children.forEach((tab, index) => {
```

Comment:
Regardless I would go for the declarative rather than imperative:
```
const tabs = props.children
const visibleTabs = tabs.slice(0, maxVisibleTabs);
const hiddenTabs =  tabs.slice(maxVisibleTabs, tabs.length);
```

Didn't test this, may be off-by-one